client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx

```diff
@@ -97,11 +113,14 @@ const GoalsStep: Step = ( { navigation } ) => {
 		recordTracksEvent( 'calypso_signup_intent_select', eventProperties );
 	};
 
-	const handleSubmit = ( submittedGoals: Onboard.SiteGoal[] ) => {
+	const handleSubmit = async ( submittedGoals: Onboard.SiteGoal[] ) => {
 		setGoals( submittedGoals );
 
 		const intent = goalsToIntent( submittedGoals );
 		setIntent( intent );
+		if ( site ) {
```

Comment:
This is just a question: in which case can this `site` be empty?

Reply:
As per this [convo](p1654095172485049-slack-CRWCHQGUB), Anchor flow doesn't have `site` during onboarding.